### PR TITLE
[SYCL][Joint Matrix] Remove duplicated matrix_multiply_ref in tests

### PR DESCRIPTION
**Pull request by YuriPlyakhin**

### CODE REVIEW

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_colmajorA_colmajorB_impl.hpp [hunk 1]
```diff
@@ -64,47 +64,29 @@ void matrix_multiply(big_matrix<T1, M, N> &C, big_matrix<T2, M, K> &A,
...
+  matrix_fill(MATRIX_M, MATRIX_N, (float *)C, 1.0f);
+  matrix_fill(MATRIX_M, MATRIX_N, (float *)D, 1.0f);
 
   big_matrix<float, MATRIX_M, MATRIX_N> MC((float *)&C);
   big_matrix<float, MATRIX_M, MATRIX_N> MD((float *)&D);
   big_matrix<bfloat16, MATRIX_M, MATRIX_K> MA((bfloat16 *)&A);
   big_matrix<bfloat16, MATRIX_K, MATRIX_N> MB((bfloat16 *)&B);
   matrix_multiply(MC, MA, MB);
-  matrix_multiply_ref(MATRIX_M, MATRIX_N, MATRIX_K);
+  matrix_multiply_ref((bfloat16 *)A, (bfloat16 *)B, (float *)D, MATRIX_M,
```
Comment (dkhaldi) [inline sycl/test-e2e/Matrix/joint_matrix_bfloat16_colmajorA_colmajorB_impl.hpp:88]: This is called using MATRIX_K

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_array_impl.hpp [hunk 2]
```diff
@@ -81,59 +81,33 @@ void matrix_multiply(big_matrix<T1, M, N> &C, big_matrix<T2, M, K> &A,
...
   big_matrix<float, MATRIX_M, MATRIX_N> MD((float *)&D);
   big_matrix<bfloat16, MATRIX_M, MATRIX_K> MA((bfloat16 *)&A);
   big_matrix<bfloat16, MATRIX_K / 2, MATRIX_N * 2> MB((bfloat16 *)&B);
 
   matrix_multiply(MC, MA, MB);
-  matrix_multiply_ref((int32_t *)A, (int32_t *)B, (int32_t *)D, MATRIX_M,
-                      MATRIX_N, MATRIX_K / 2);
+  matrix_multiply_ref<bfloat16, bfloat16, float, 2>(
+      (bfloat16 *)A, (bfloat16 *)B, (float *)D, MATRIX_M, MATRIX_N,
+      MATRIX_K / 2);
```
Comment (dkhaldi) [inline sycl/test-e2e/Matrix/joint_matrix_bfloat16_array_impl.hpp:110]: this is called using MATRIX_K/2

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_colmajorA_colmajorB_impl.hpp [hunk 3]
```diff
@@ -64,47 +64,29 @@ void matrix_multiply(big_matrix<T1, M, N> &C, big_matrix<T2, M, K> &A,
...
 
-static constexpr size_t MATRIX_M = TM * 2;
-static constexpr size_t MATRIX_N = TN * 2;
-static constexpr size_t MATRIX_K = TK * 2;
-bfloat16 A[MATRIX_K][MATRIX_M];
-bfloat16 B[MATRIX_N][MATRIX_K];
-float C[MATRIX_M][MATRIX_N];
-float D[MATRIX_M][MATRIX_N];
-
-void matrix_multiply_ref(int M, int N, int K) {
```
Comment (dkhaldi) [inline sycl/test-e2e/Matrix/joint_matrix_bfloat16_colmajorA_colmajorB_impl.hpp:75]: this matrix_ref is different from the one above.
The one above uses matrix B that is VNNIed, so it uses two successive elements of B.
This one does not.
So I don't thin we can use same ref function for both.

Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/joint_matrix_bfloat16_colmajorA_colmajorB_impl.hpp:75]: Not sure, I understand the comment. It seems to me the multiplication is equivalent in this test case.

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_colmajorA_colmajorB_impl.hpp (hunk 1 above)
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/joint_matrix_bfloat16_colmajorA_colmajorB_impl.hpp:88]: What is your suggestion? Not sure, I understand the comment....

File: sycl/test-e2e/Matrix/joint_matrix_bfloat16_array_impl.hpp (hunk 2 above)
Comment (YuriPlyakhin) [reply sycl/test-e2e/Matrix/joint_matrix_bfloat16_array_impl.hpp:110]: Yes, this is what is in the code. Not sure what is this comment about...

Comment (dkhaldi) [reply sycl/test-e2e/Matrix/joint_matrix_bfloat16_array_impl.hpp:110]: I see now you are playing with VF value to rule out the VNNI case. That makes sense.

**Review by dkhaldi — APPROVED** [review]
LGTM

**Comment by YuriPlyakhin** [comment]
@intel/llvm-reviewers-runtime , could someone please approve? I don't know why Dounia's approval is not enough...

**Comment by aelovikov-intel** [comment]
> @intel/llvm-reviewers-runtime , could someone please approve? I don't know why Dounia's approval is not enough...

Can you try merging origin/sycl?

**Comment by aelovikov-intel** [comment]
> @intel/llvm-reviewers-runtime , could someone please approve? I don't know why Dounia's approval is not enough...

https://github.com/intel/llvm/pull/11650 should fix it.

**Comment by YuriPlyakhin** [comment]
@intel/llvm-gatekeepers , all green, please merge.

**Pull request merged**